### PR TITLE
Remove runtime dependency checks when loading plugins

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Currently all runtime dependencies must be available in order to load a plugin. With plugins now covering multiple technologies this doesn't really make sense and all our users currently disable this feature so it makes sense to remove this completely and let users check for runtime dependencies on demand.

Fixes: #423

### CODE REVIEW

**Comment by tbzatek** [comment]
Having both `bd_X_check_deps()` and `bd_X_is_tech_avail()` in place is quite confusing so I'm for simplification of the API. The `bd_X_is_tech_avail()` should cover most of the checking cases.

These are mostly runtime dependencies (somewhat optional), the linking deps are perhaps more important.

**Review by tbzatek — APPROVED** [review]
Looks good!

File: tests/btrfs_test.py [hunk 1]
```diff
@@ -36,6 +35,9 @@ def setUpClass(cls):
         else:
             BlockDev.reinit(cls.requested_plugins, True, None)
 
+
+class BtrfsTestCase(BtrfsTest):
```
Comment (vpodzime) [inline tests/btrfs_test.py:39]: I think these deserve a better name. If someone goes and adds a new test here, they can easily use `BtrfsTest` as the base class.

File: docs/3.0-api-changes.xml [hunk 2]
```diff
@@ -5,6 +5,16 @@
   This is a list of API changes in libblockdev 3.0 with suggested replacements.
   </para>
 
+  <simplesect><title>Library</title>
+  <para>
+    <literal>bd_switch_init_checks()</literal> has been removed. Checks for runtime dependencies during libblockdev initialization are now disabled and plugins will be loaded even if a runtime dependency is missing.
```
Comment (vpodzime) [inline docs/3.0-api-changes.xml:10]: I'd make this **bold** because this is a quite a big change in the default behavior.

**Review by vpodzime — APPROVED** [review]
Looks good to me otherwise.

File: src/python/gi/overrides/BlockDev.py [hunk 3]
```diff
@@ -1123,6 +1123,12 @@ def crypto_luks_uuid(device):
 __all__.append("crypto_luks_uuid")
 
 
+def switch_init_checks(enable):
+    # no-op, init checks are no longer supported
+    return
```
[the remark this answers is not in the document]
Comment (vojtechtrefny) [reply src/python/gi/overrides/BlockDev.py:1128]: This is just a temporary function, I will remove it after releasing 3.0. This is here just to make sure blivet tests in our CI won't start failing after merging this. (There is a comment explaining this above, there are two more functions/overrides like this.)

File: tests/btrfs_test.py (hunk 1 above)
Comment (vojtechtrefny) [reply tests/btrfs_test.py:39]: I plan to do a big test "cleanup" before the 3.0 release, there is a lot of outdated/messy things there. I kept this simple, we have `XXXTestCase` and `XXXTest` for other plugins too.

File: docs/3.0-api-changes.xml (hunk 2 above)
Comment (vojtechtrefny) [reply docs/3.0-api-changes.xml]: Ok. Big changes like these will be documented in release notes too.

**Pull request merged**